runtime/lua/vim/secure.lua

```diff
@@ -21,6 +21,50 @@ local function read_trust()
   return trust
 end
 
+--- Reads the given file/directory at {fullpath} and returns the
+--- contents and a hash of the contents. If {fullpath} is not a directory
+--- and {bufnr} is given, then contents are read from the {bufnr} buffer rather
+--- than the filesytesm.
+---
+---@param fullpath (string) Path to a file or directory to read.
+---@param bufnr (number?) The number of the buffer.
+---@return string|boolean? contents the contents of the file, or true if it's a directory
+---@return string? hash the hash of the contents, or "directory" if it's a directory
+local function read_and_hash(fullpath, bufnr)
```

Comment:
Nit: rename this function just `hash` or something: if it's a directory then nothing is being read, which is misleading (update the docstring accordingly as well).

Reply:
Actually if we go with my suggestion [here](https://github.com/neovim/neovim/pull/33617/files#r2063888847) then ignore this suggestion, since we would be reading the contents of the directory.

(I personally find short, single word function names in Lua more idiomatic/preferable, but that's a subjective/stylistic preference so won't push on it).